crates/bdk/src/keys/mod.rs

```diff
@@ -20,6 +20,8 @@ use core::marker::PhantomData;
 use core::ops::Deref;
 use core::str::FromStr;
 
+use rand_core::RngCore;
```

Comment:
We might as well restrict to crytographic-secure PRNG with [`CryptoRng`](https://docs.rs/rand_core/latest/rand_core/trait.CryptoRng.html) trait instead?